### PR TITLE
add documentation and capture more common code

### PR DESCRIPTION
**Pull request by olasaadi99**

### CODE REVIEW

**Review by adisos — COMMENTED** [review]
1. Can more similar parser code be captured? 
such as in prasers `getSGconfig`, and in `VPCConfigsFromResources`, and maybe more functions.

**Review by adisos — COMMENTED** [review]
please also add documentation to nacl_analysis

**Comment by olasaadi99** [comment]
> 1. VPCConfigsFromResources

`getSGconfig` yes, but `VPCConfigsFromResources` I prefer not to do, because it is not the same resources for both providers yet, and also not the same arguments for each getConfig function for both providers

File: pkg/ibmvpc/nacl_analysis.go [hunk 1]
```diff
@@ -92,26 +95,9 @@ func (na *IBMNACLAnalyzer) GetNACLRule(index int) (ruleStr string, ruleRes *comm
...
-		ruleObj.Index = index
-		if isIngress {
-			ingressRules = append(ingressRules, ruleObj)
-		} else {
-			egressRules = append(egressRules, ruleObj)
-		}
-	}
-	return ingressRules, egressRules, nil
+	ingressRules, egressRules, referencedIPblocks, err := commonvpc.GetNACLRules(na.GetNumberOfRules(), na)
+	na.referencedIPblocks = referencedIPblocks
```
Comment (haim-kermany) [inline pkg/ibmvpc/nacl_analysis.go:101]: SpecificNACLAnalyzer should implement SetReferencedIPblocks()
SetReferencedIPblocks() should be called inside commonvpc.GetNACLRules()

File: pkg/commonvpc/parser.go [hunk 2]
```diff
@@ -376,3 +388,34 @@ func GetSubnetsNodes(subnets []*Subnet) []vpcmodel.Node {
 	}
 	return res
 }
+
+// UpdateConfigWithSGAndPrepareAnalyzer updates config with sg layer results and prepare analyzer
+func UpdateConfigWithSGAndPrepareAnalyzer(res *vpcmodel.MultipleVPCConfigs, sgLists map[string][]*SecurityGroup,
```
Comment (haim-kermany) [inline pkg/commonvpc/parser.go:393]: split this function to two functions, give each of function a proper name.

File: pkg/commonvpc/sg_analysis.go [hunk 3]
```diff
@@ -292,3 +297,24 @@ func GetIPBlockResult(cidr, address, name *string,
 	}
 	return ipBlock, cidrRes, nil
 }
+
+// GetSGRules returns ingress and egress rule objects
+func GetSGRules(numRules int, sga SpecificSGAnalyzer) (ingressRules, egressRules []*SGRule, err error) {
```
Comment (haim-kermany) [inline pkg/commonvpc/sg_analysis.go:302]: parameter numRules is not necessary

File: pkg/awsvpc/nacl_analysis.go [hunk 4]
```diff
@@ -88,25 +91,12 @@ func (na *AWSNACLAnalyzer) GetNACLRule(index int) (ruleStr string, ruleRes *comm
 	return ruleStr, ruleRes, isIngress, nil
 }
 
+// SetReferencedIPblocks updates referenced ip blocks
+func (na *AWSNACLAnalyzer) SetReferencedIPblocks(referencedIPblocks []*ipblock.IPBlock) {
+	na.referencedIPblocks = referencedIPblocks
```
Comment (haim-kermany) [inline pkg/awsvpc/nacl_analysis.go:96]: put SetReferencedIPblocks() next to ReferencedIPblocks()

**Review by haim-kermany — APPROVED** [review]

**Pull request merged**